File: AgoraChat/Code/Custom/View/ACDNaviCustomView.m
//
//  ACDNaviCustomView.m
//  ChatDemo-UI3.0
//
//  Created by liang on 2021/10/21.
//  Copyright © 2021 easemob. All rights reserved.
//

#import "ACDNaviCustomView.h"
@interface ACDNaviCustomView ()

@end

@implementation ACDNaviCustomView
- (void)prepare {
    [self addSubview:self.titleImageView];
    [self addSubview:self.addButton];
}


- (void)placeSubViews {
    [self.titleImageView mas_makeConstraints:^(MASConstraintMaker *make) {
        make.top.equalTo(self).offset(kAgroaPadding * 4.4);
        make.centerX.equalTo(self);
        make.bottom.equalTo(self).offset(-kAgroaPadding);
    }];
    
    [self.addButton mas_makeConstraints:^(MASConstraintMaker *make) {
        make.centerY.equalTo(self.titleImageView);
        make.right.equalTo(self).offset(-kAgroaPadding);
        make.size.equalTo(@40.0);
    }];
}

#pragma mark action
- (void)addAction {
    if (self.addActionBlock) {
        self.addActionBlock();
    }
}

#pragma mark getter and setter
- (UIButton *)addButton {
    if (_addButton == nil) {
        _addButton = [[UIButton alloc] initWithFrame:CGRectMake(0, 0, 8, 15)];
        _addButton.contentMode = UIViewContentModeScaleAspectFill;
        [_addButton setImage:ImageWithName(@"contact_add_contacts") forState:UIControlStateNormal];
        [_addButton addTarget:self action:@selector(addAction) forControlEvents:UIControlEventTouchUpInside];
    }
    return _addButton;
}

- (UIImageView *)titleImageView {
    if (_titleImageView == nil) {
        _titleImageView = [[UIImageView alloc] init];
        _titleImageView.contentMode = UIViewContentModeScaleAspectFill;
        _titleImageView.image = ImageWithName(@"nav_title_contacts");
    }
    return _titleImageView;
}


@end

File: AgoraChat-Swift/AgoraChat-Swift/code/Custom/View/ACDNaviCustomView.swift
//
//  ACDNaviCustomView.swift
//  AgoraChat-Swift
//
//  Created by 冯钊 on 2022/10/18.
//

import UIKit
import SnapKit

class ACDNaviCustomView: UIView {

    let addButton = UIButton(type: .custom)
    let titleImageView = UIImageView()
    
    var addActionHandle: (() -> Void)?
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        
        self.addButton.contentMode = .scaleAspectFill
        self.addButton.setImage(UIImage(named: "contact_add_contacts"), for: .normal)
        self.addButton.addTarget(self, action: #selector(addAction), for: .touchUpInside)
        self.addSubview(self.addButton)
        
        self.titleImageView.contentMode = .scaleAspectFill
        self.titleImageView.image = UIImage(named: "nav_title_contacts")
        self.addSubview(self.titleImageView)
        
        self.titleImageView.snp.makeConstraints { make in
            make.height.equalTo(20)
            make.centerX.equalTo(self)
            make.bottom.equalTo(self).offset(-10)
        }
        self.addButton.snp.makeConstraints { make in
            make.centerY.equalTo(self.titleImageView)
            make.right.equalTo(self).offset(-10)
            make.size.equalTo(40)
        }
    }
    
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    @objc private func addAction() {
        self.addActionHandle?()
    }
}
